### PR TITLE
tls_codec: implement `U24`

### LINKED ISSUE #1277: tls_codec: no support for `uint24_t`-prefixed variable length vectors
`tls_codec` is missing the `TlsByteSliceU24` and `TlsByteVecU24` types. These are useful in verifying Signed Certificate Timestamps, which contain some uint24 sized fields [in the`digitally-signed` component](https://datatracker.ietf.org/doc/html/rfc6962#section-3.2).

I've worked around this in `sigstore-rs` [by hacking together a partial implementation of `TlsByteVecU24`](https://github.com/trail-of-forks/sigstore-rs/blob/bf01ea268773a237869b5906623326dd7a6fe9a4/src/crypto/transparency.rs#L114-L152), but it'd be nice to have a proper one upstream. I'm more than happy to contribute this myself if it'd be helpful.

### PR DESCRIPTION
**Pull request by tnytown**
Resolves #1277.

### CODE REVIEW

File: tls_codec/src/primitives.rs [hunk 1]
```diff
@@ -124,7 +124,7 @@ macro_rules! impl_unsigned {
         impl DeserializeBytes for $t {
             #[inline]
             fn tls_deserialize_bytes(bytes: &[u8]) -> Result<(Self, &[u8]), Error> {
-                let len = core::mem::size_of::<$t>();
+                let len = <$t>::default().tls_serialized_len();
```
[the remark this answers is not in the document]
Comment (baloo) [reply tls_codec/src/primitives.rs:127]: This was a `const-fn`, this isn't anymore.

File: tls_codec/src/tls_vec.rs [hunk 2]
```diff
@@ -42,7 +42,7 @@ macro_rules! impl_byte_deserialize {
         #[cfg(feature = "std")]
         #[inline(always)]
         fn deserialize_bytes<R: Read>(bytes: &mut R) -> Result<Self, Error> {
-            let len = <$size>::tls_deserialize(bytes)? as usize;
+            let len = <$size>::tls_deserialize(bytes)?.try_into().unwrap();
```
[the remark this answers is not in the document]
Comment (franziskuskiefer) [reply tls_codec/src/tls_vec.rs:45]: The conversion is probably fine. The `unwrap` may slow down things. Shall we return an error instead?
There are benchmarks that could tell us if this really slows down things. Can you use them to see if you see any significant issue here?

File: tls_codec/src/tls_vec.rs [hunk 3]
```diff
@@ -885,15 +885,18 @@ macro_rules! impl_tls_byte_vec {
...
 
 impl_tls_byte_vec!(u8, TlsByteVecU8, 1);
 impl_tls_byte_vec!(u16, TlsByteVecU16, 2);
+impl_tls_byte_vec!(U24, TlsByteVecU24, 3);
 impl_tls_byte_vec!(u32, TlsByteVecU32, 4);
 
 // Secrets should be put into these Secret tls vectors as they implement zeroize.
 impl_secret_tls_vec!(u8, SecretTlsVecU8, 1);
 impl_secret_tls_vec!(u16, SecretTlsVecU16, 2);
+impl_secret_tls_vec!(U24, SecretTlsVecU24, 2);
```
Comment (franziskuskiefer) [inline tls_codec/src/tls_vec.rs:899]: Should this be `3`?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Sorry for the slow review. Holidays ... 🎄🤶🏻
Generally looks good to me. Two nits then we can get this in.
I quickly checked RFC 9000 but it doesn't seem to allow u24 for variable length encodings but maybe double check that we shouldn't add it for variable length encodings too.

File: tls_codec/src/tls_vec.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (franziskuskiefer) [reply tls_codec/src/tls_vec.rs:45]: Cool, looks like this shold be fine then :)

**Review by franziskuskiefer — APPROVED** [review]
Thanks, lets get this in then :)

**Pull request merged**